workers/data_refinery_workers/processors/smashing_utils.py

```diff
@@ -643,18 +667,22 @@ def write_non_data_files(job_context: Dict) -> Dict:
 
     # Write samples metadata to TSV
     try:
-        tsv_paths = write_tsv_json(job_context)
-        job_context['metadata_tsv_paths'] = tsv_paths
+        write_tsv_json(job_context)
         # Metadata to JSON
         job_context['metadata']['created_at'] = timezone.now().strftime('%Y-%m-%dT%H:%M:%S')
-        with open(job_context["output_dir"] + 'aggregated_metadata.json',
-                  'w',
-                  encoding='utf-8') as metadata_file:
+        aggregated_metadata_path = os.path.join(job_context["output_dir"],
+                                                'aggregated_metadata.json')
+        with open(aggregated_metadata_path, 'w', encoding='utf-8') as metadata_file:
             json.dump(job_context['metadata'], metadata_file, indent=4, sort_keys=True)
+
+        if job_context['filtered_samples']:
+            # generate filtered samples file only if some samples were skipped
+            filtered_samples_path = os.path.join(job_context["output_dir"],
+                                                 'filtered_samples_metadata.json')
+            with open(filtered_samples_path, 'w',encoding='utf-8') as metadata_file:
+                json.dump(job_context['filtered_samples'], metadata_file, indent=4, sort_keys=True)
```

Comment:
I'm not sure this metadata file is sufficient. For one thing, we tend to include a JSON version and a TSV version of every metadata file we include in the compendia, in case a user is only comfortable with one of the formats.

Secondly, in https://github.com/AlexsLemonade/refinebio/issues/1838#issuecomment-559143625 you mention

> We could also add all sample metadata there, and the experiment relations.

I would think we probably want to do this because if a user is actually using this file then they may be interested in analyzing why some samples got filtered. Would you agree with this @jaclyn-taroni? Should we include every metadata field we have about every sample that got filtered including it's experiment_accession_code?

Reply:
> we tend to include a JSON version and a TSV version of every metadata file we include in the compendia, in case a user is only comfortable with one of the formats.

We only have a tsv version for the samples inside the compendia. The main `aggregated_metadata.json` file doesn't have a tsv version.

![image](https://user-images.githubusercontent.com/1882507/70456498-37a44200-1a7c-11ea-90c1-4a6602566218.png)

I'd say the new `filtered_samples_metadata.json` file should be similar to `aggregated_metadata.json`. Do you think this makes sense?

> Should we include every metadata field we have about every sample that got filtered including it's experiment_accession_code?

Ok, I'll also include this information for each sample.